cmd/drand-cli/cli.go

```diff
@@ -848,35 +848,44 @@ func deleteBeaconCmd(c *cli.Context) error {
 		return err
 	}
 
+	var er error
 	for beaconID, storePath := range stores {
-		store, err := boltdb.NewBoltStore(path.Join(storePath, core.DefaultDBFolder), conf.BoltOptions())
-		if err != nil {
-			return fmt.Errorf("beacon id [%s] - invalid bolt store creation: %s", beaconID, err)
-		}
-		defer store.Close()
-
-		lastBeacon, err := store.Last()
-		if err != nil {
-			return fmt.Errorf("beacon id [%s] - can't fetch last beacon: %s", beaconID, err)
-		}
-		if startRound > lastBeacon.Round {
-			return fmt.Errorf("beacon id [%s] - given round is ahead of the chain: %d", beaconID, lastBeacon.Round)
-		}
-		if c.IsSet(verboseFlag.Name) {
-			fmt.Printf("beacon id [%s] -  planning to delete %d beacons \n", beaconID, (lastBeacon.Round - startRound))
+		if er != nil {
+			return er
 		}
+		// Using an anonymous function to not leak the defer
+		er = func() error {
+			store, err := boltdb.NewBoltStore(path.Join(storePath, core.DefaultDBFolder), conf.BoltOptions())
```

Comment:
My understanding is that these are all accessing different DBs: one for each "beaconID", so it should be fine. (It's part of these new "multi-beacon/multi-frequency" features).

So as I understand it we are storing each "beacon DB" in its own store instead of mashing them all in the same DB.